Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPickerDropdown.cs

```diff
@@ -52,6 +52,11 @@ protected override void OnDestroy()
             m_RebindingOperation = null;
         }
 
+        public void Dispose()
+        {
+            m_RebindingOperation?.Dispose();
+        }
```

Comment:
Should these ones that now have IDisposable on them implement a full dispose pattern with a finalizer?

Reply:
Good question.

I _think_ we don't need to. Reading https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose, it seems that for the IDisposable pattern we should implement a Dispose(bool), where the bool indicates if Dispose was called from a finalizer or directly. If it was called from a finalizer, we should only dispose our own native data, but not call dispose on other managed objects (as they would get their own finalizer called) - which in this case would mean the finalizer would do nothing.